gradle/swagger.gradle

```diff
@@ -12,33 +12,59 @@ buildscript {
     }
 }
 
-openApiGenerate {
-    generatorName = "spring"
-    inputSpec = "$rootDir/src/main/resources/swagger/api.yml".toString()
-    outputDir = "$buildDir/openapi".toString()
-    apiPackage = "com.icthh.xm.tmf.ms.customer.api"
-    modelPackage = "com.icthh.xm.tmf.ms.customer.model"
-    apiFilesConstrainedTo = [""]
-    modelFilesConstrainedTo = [""]
-    supportingFilesConstrainedTo = ["ApiUtil.java"]
-    configOptions = [delegatePattern: "true", title: "customer"]
-    validateSpec = true
-    importMappings = [Problem:"org.zalando.problem.Problem"]
+void createOpenApiGenerateTask(String ymlName) {
+    String taskName = "openApiGenerate_" + ymlName;
+    task(taskName, type: org.openapitools.generator.gradle.plugin.tasks.GenerateTask) {
+        generatorName = "spring"
+        inputSpec = "$rootDir/src/main/resources/swagger/${ymlName}.yml".toString()
+        outputDir = "$buildDir/openapi".toString()
+        apiPackage = "com.icthh.xm.tmf.ms.customer.api"
+        modelPackage = "com.icthh.xm.tmf.ms.customer.model"
+        apiFilesConstrainedTo = [""]
+        modelFilesConstrainedTo = [""]
+        supportingFilesConstrainedTo = ["ApiUtil.java"]
+        configOptions = [delegatePattern: "true", title: "customer"]
+        validateSpec = true
+        importMappings = [Problem: "org.zalando.problem.Problem"]
+    }
+    compileJava.dependsOn(taskName)
```

Comment:
seems like this is a duplication for `createOpenApiGenerateTaskV31`
is it possible to refactor to one method?

Reply:
@amedvedchuk one method was created